src/auto-evo/simulation/food_source/MarineSnowFoodSource.cs

```diff
@@ -19,14 +19,15 @@ public MarineSnowFoodSource(Patch patch)
         }
     }
 
-    public override float FitnessScore(Species species)
+    public override float FitnessScore(Species species, SimulationCache simulationCache)
     {
         var microbeSpecies = (MicrobeSpecies)species;
 
         var predatorSpeed = microbeSpecies.BaseSpeed;
-        predatorSpeed += ProcessSystem
-            .ComputeEnergyBalance(microbeSpecies.Organelles.Organelles, patch.Biome,
-                microbeSpecies.MembraneType).FinalBalance;
+
+        var energyBalance = simulationCache.GetEnergyBalanceForSpecies(microbeSpecies, patch);
+
+        predatorSpeed += energyBalance.FinalBalance;
```

Comment:
```suggestion
        predatorSpeed += energyBalance.FinalBalance;
```

personally I'd group these related statements

Reply:
I wouldn't because the energy balance is also used a few lines later. If anything this line should be swapped with 26.